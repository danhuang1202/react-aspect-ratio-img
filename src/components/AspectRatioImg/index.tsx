import React, { FunctionComponent, ReactNode } from 'react'
import { AspectRatioFence } from 'react-aspect-ratio-fence'
import 'react-aspect-ratio-fence/css/components/AspectRatioFence/style.css'

// @ts-ignore
import styles from './style.css'

type Props = {
  /**
   *  outter html tag name
   *  @default div
   * */
  outerElementType?: string
  /** Custom class name */
  customClass?: string
  /**
   * The aspect ratio of an image describes the proportional
   * relationship between width and height.
   *  @default 1
   **/
  ratio: number
  /** The src attribute of <img /> tag */
  src: string
  /** Attributes of <img /> tag */
  imgAttributes?: object
  /** custom children node */
  children?: ReactNode
}

const AspectRatioImg: FunctionComponent<Props> = ({
  outerElementType = 'div',
  ratio = 1,
  src,
  imgAttributes,
  customClass,
  children
}) => {
  return (
    <AspectRatioFence
      elementType={outerElementType}
      customClass={customClass}
      ratio={ratio}
    >
      <img alt="" {...imgAttributes} src={src} className={styles.img} />
      {children}
    </AspectRatioFence>
  )
}

export default AspectRatioImg
