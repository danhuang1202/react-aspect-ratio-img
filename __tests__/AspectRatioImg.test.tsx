import React from 'react'
import renderer from 'react-test-renderer'
import AspectRatioImg from '@components/AspectRatioImg'

describe('[snapshot] AspectRatioImg', () => {
  const defaultProps = {
    ratio: 16 / 9,
    src: 'aspect-ratio-640w.jpg',
    imgAttributes: {
      alt: 'AspectRatioImg',
      srcSet: `
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x`
    }
  }

  it('AspectRatioImg', () => {
    const dom = renderer.create(<AspectRatioImg {...defaultProps} />)
    expect(dom).toMatchSnapshot()
  })

  it('AspectRatioImg with outerElementType', () => {
    const props = {
      ...defaultProps,
      outerElementType: 'span'
    }

    const dom = renderer.create(<AspectRatioImg {...props} />)
    expect(dom).toMatchSnapshot()
  })
})
