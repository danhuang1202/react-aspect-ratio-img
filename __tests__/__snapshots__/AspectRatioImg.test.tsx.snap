// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] AspectRatioImg AspectRatioImg 1`] = `
<div
  className="AspectRatioFence_wrap_2HH2L"
  style={
    Object {
      "--ratio": "(1.7777777777777777)",
    }
  }
>
  <img
    alt="AspectRatioImg"
    className="AspectRatioImg_img_-WkaR"
    src="aspect-ratio-640w.jpg"
    srcSet="
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x"
  />
</div>
`;

exports[`[snapshot] AspectRatioImg AspectRatioImg with outerElementType 1`] = `
<span
  className="AspectRatioFence_wrap_2HH2L"
  style={
    Object {
      "--ratio": "(1.7777777777777777)",
    }
  }
>
  <img
    alt="AspectRatioImg"
    className="AspectRatioImg_img_-WkaR"
    src="aspect-ratio-640w.jpg"
    srcSet="
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x"
  />
</span>
`;
